File: app/src/main/java/com/door43/translationstudio/ui/translate/review/OnReviewModeListener.java
package com.door43.translationstudio.ui.translate.review;

import android.widget.TextView;

import com.door43.translationstudio.ui.spannables.NoteSpan;
import com.door43.translationstudio.ui.translate.OnAdapterListener;
import com.door43.translationstudio.ui.translate.ReviewListItem;
import com.door43.translationstudio.ui.translate.TranslationHelp;

import org.unfoldingword.resourcecontainer.Link;

public interface OnReviewModeListener extends OnAdapterListener {
    void onNoteClick(TranslationHelp note, int resourceCardWidth);
    void onWordClick(String resourceContainerSlug, Link word, int resourceCardWidth);
    void onQuestionClick(TranslationHelp question, int resourceCardWidth);
    void onResourceTabNotesSelected(ReviewHolder holder);
    void onResourceTabWordsSelected(ReviewHolder holder);
    void onResourceTabQuestionsSelected(ReviewHolder holder);
    void onTapResourceCard();
    void onNotifyItemChanged(int position);
    void onEditorToggle(ReviewHolder holder);
    void onApplyChangedText(CharSequence s, ReviewHolder holder);
    void onUndoTextInTarget(ReviewHolder holder);
    void onRedoTextInTarget(ReviewHolder holder);
    void onDoneSwitchClicked(ReviewHolder holder, boolean checked);
    void onCreateFootnoteAtSelection(ReviewHolder holder);
    CharSequence onRenderSourceText(ReviewListItem item);
    void onSearchItemUpdated(int position, TextView view, boolean isTarget);
    void onMergeConflictItemCancel(int position);
    void onMergeConflictItemConfirm(int position);
    CharSequence onRenderTargetText(ReviewHolder holder, ReviewListItem item, boolean editable);
    CharSequence onRenderTargetText(ReviewHolder holder, ReviewListItem item);
    void onAddMissingVerses(int position);
    void onRenderHelps(ReviewListItem item);
    void onSourceFootnoteClick(ReviewListItem item, NoteSpan span, int start, int end);
}

File: app/src/main/java/com/door43/translationstudio/ui/translate/review/OnReviewModeListener.kt
package com.door43.translationstudio.ui.translate.review

import android.view.View
import android.widget.TextView
import com.door43.translationstudio.rendering.model.TextNode
import com.door43.translationstudio.ui.translate.OnAdapterListener
import com.door43.translationstudio.ui.translate.ReviewListItemOld
import com.door43.translationstudio.ui.translate.TranslationHelp
import org.unfoldingword.resourcecontainer.Link

interface OnReviewModeListener : OnAdapterListener {
    fun onNoteClick(note: TranslationHelp, resourceCardWidth: Int)
    fun onWordClick(resourceContainerSlug: String, word: Link, resourceCardWidth: Int)
    fun onQuestionClick(question: TranslationHelp, resourceCardWidth: Int)
    fun onResourceTabNotesSelected(holder: ReviewHolder)
    fun onResourceTabWordsSelected(holder: ReviewHolder)
    fun onResourceTabQuestionsSelected(holder: ReviewHolder)
    fun onTapResourceCard()
    fun onNotifyItemChanged(position: Int)
    fun onEditorToggle(holder: ReviewHolder)
    fun onApplyChangedText(s: CharSequence, holder: ReviewHolder)
    fun onUndoTextInTarget(holder: ReviewHolder)
    fun onRedoTextInTarget(holder: ReviewHolder)
    fun onDoneSwitchClicked(holder: ReviewHolder, checked: Boolean)
    fun onCreateFootnoteAtSelection(holder: ReviewHolder)
    fun onRenderSourceText(item: ReviewListItemOld): List<TextNode>
    fun onSearchItemUpdated(position: Int, view: TextView, isTarget: Boolean)
    fun onMergeConflictItemCancel(position: Int)
    fun onMergeConflictItemConfirm(position: Int)
    fun onRenderTargetText(
        holder: ReviewHolder,
        item: ReviewListItemOld,
        editable: Boolean
    ): List<TextNode>

    fun onRenderTargetText(holder: ReviewHolder, item: ReviewListItemOld): List<TextNode>
    fun onAddMissingVerses(holder: ReviewHolder)
    fun onRenderHelps(item: ReviewListItemOld)
    fun onSourceNoteClick(item: ReviewListItemOld, marker: TextNode.NoteMarker)
    fun onNoteClick(holder: ReviewHolder, item: ReviewListItemOld, marker: TextNode.NoteMarker, start: Int, end: Int, editable: Boolean)
    fun onVerseClick(item: ReviewListItemOld, marker: TextNode.VerseMarker)
    fun onVerseLongClick(view: View, holder: ReviewHolder, item: ReviewListItemOld, marker: TextNode.VerseMarker, start: Int, end: Int)
}
